modules/apps/frontend-js/frontend-js-collapse-support-web/src/main/resources/META-INF/resources/CollapseProvider.js

```diff
@@ -196,7 +197,7 @@ class CollapseProvider {
 	}
 
 	_onTriggerClick = (event) => {
-		const trigger = event.delegateTarget;
+		const trigger = event.target.closest(Selector.TRIGGER);
```

Comment:
Hey @kresimir-coko, why don't you add this to our `delegate` function once and for all to avoid doing this?

Stick it to the [`event`](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js#L48) if you wish so it'll be ready and we keep the API the same. It's not like `delegate` is a standard API anyways!

If we don't want to monkey patch `DOMEvent` because that's evil, then we can pass an additional argument to the callback or our own Event object, but this doesn't feel like a very usable API at this point.

Reply:
@jbalsas Sure, I can do that 👍

Reply:
> If we don't want to monkey patch `DOMEvent` because that's evil

Evil indeed, but I'd actually be more worried about what might happen if this ever got used in a context where we were interoperating with some other abstraction that put a facade over real objects, because at that point you've lost control. (Example: React uses "synthetic" event objects and furthermore recycles and reuses these event objects from a pool, so you can't reliably monkey patch those at all, because their identity can and does change even when you think you're holding a stable reference to one.)

It is unlikely to bite us, I think, but I just wanted to mention it.

Reply:
Yeah, I'd expect this to be used only in contexts outside of `react`... and become marginal the more we migrate into it.. 🙏